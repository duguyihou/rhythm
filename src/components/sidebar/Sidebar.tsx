import Link from 'next/link'

import { Logo } from 'components/logo'
import { Player, usePlayerStore } from 'features/player'

import SideNavigation from './SideNavigation'

const Sidebar = () => {
  const { currentPlaying } = usePlayerStore()
  return (
    <div className="hidden md:flex md:flex-shrink-0">
      <div className="flex flex-col w-64">
        <div className="flex flex-col h-0 flex-1">
          <div className="flex items-center h-16 flex-shrink-0 px-4">
            <Link href={'/'}>
              <div className="cursor-pointer">
                <Logo />
              </div>
            </Link>
          </div>
          <div className="flex-1 flex flex-col overflow-y-auto">
            <nav className="flex-1 px-2 py-4 space-y-1">
              <SideNavigation />
            </nav>
            <section className="mb-10">
              {currentPlaying && <Player station={currentPlaying} />}
            </section>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Sidebar
