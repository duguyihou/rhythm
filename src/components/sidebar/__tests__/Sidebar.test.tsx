import { render } from '@testing-library/react'

import Sidebar from '../Sidebar'

test('sidebar', () => {
  const sidebar = render(<Sidebar />)

  expect(sidebar).toMatchSnapshot()
})
