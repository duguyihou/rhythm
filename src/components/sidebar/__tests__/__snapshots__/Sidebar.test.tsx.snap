// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hidden md:flex md:flex-shrink-0"
      >
        <div
          class="flex flex-col w-64"
        >
          <div
            class="flex flex-col h-0 flex-1"
          >
            <div
              class="flex items-center h-16 flex-shrink-0 px-4"
            >
              <div
                class="cursor-pointer"
              >
                <svg
                  fill="none"
                  height="48px"
                  viewBox="0 0 48 48"
                  width="48px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="white"
                    fill-opacity="0.01"
                    height="48"
                    width="48"
                  />
                  <circle
                    cx="24"
                    cy="22"
                    fill="#2F88FF"
                    r="4"
                    stroke="black"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                  <path
                    d="M21.5238 33.0539C22.7619 31.6487 25.2381 31.6486 26.4762 33.0539C27.7143 34.4592 26.4763 43.3608 25.6508 44.2975C24.8254 45.2342 23.1746 45.2342 22.3492 44.2975C21.5238 43.3608 20.2857 34.4591 21.5238 33.0539Z"
                    fill="#2F88FF"
                    stroke="black"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                  <path
                    d="M30.9668 30.513C33.4289 28.4957 35 25.4313 35 22C35 15.9249 30.0751 11 24 11C17.9249 11 13 15.9249 13 22C13 25.4313 14.5711 28.4957 17.0332 30.513"
                    stroke="black"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                  <path
                    d="M31.9258 38.1656C37.8928 35.2345 42 29.0969 42 22C42 12.0589 33.9411 4 24 4C14.0589 4 6 12.0589 6 22C6 29.0969 10.1072 35.2345 16.0742 38.1656"
                    stroke="black"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="4"
                  />
                </svg>
              </div>
            </div>
            <div
              class="flex-1 flex flex-col overflow-y-auto"
            >
              <nav
                class="flex-1 px-2 py-4 space-y-1"
              >
                <div
                  class="text-gray-700 hover:bg-gray-300 hover:text-gray-900 cursor-pointer group flex items-center px-2 py-2 text-base font-medium rounded-md"
                >
                  <div
                    class="w-4 h-4"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-circle-play "
                      data-icon="circle-play"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M512 256c0 141.4-114.6 256-256 256S0 397.4 0 256S114.6 0 256 0S512 114.6 512 256zM188.3 147.1c-7.6 4.2-12.3 12.3-12.3 20.9V344c0 8.7 4.7 16.7 12.3 20.9s16.8 4.1 24.3-.5l144-88c7.1-4.4 11.5-12.1 11.5-20.5s-4.4-16.1-11.5-20.5l-144-88c-7.4-4.5-16.7-4.7-24.3-.5z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                  <div
                    class="px-2"
                  >
                    Listen Now
                  </div>
                </div>
                <div
                  class="text-gray-700 hover:bg-gray-300 hover:text-gray-900 cursor-pointer group flex items-center px-2 py-2 text-base font-medium rounded-md"
                >
                  <div
                    class="w-4 h-4"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-border-all "
                      data-icon="border-all"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M384 96V224H256V96H384zm0 192V416H256V288H384zM192 224H64V96H192V224zM64 288H192V416H64V288zM64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H384c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                  <div
                    class="px-2"
                  >
                    Browse
                  </div>
                </div>
              </nav>
              <section
                class="mb-10"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hidden md:flex md:flex-shrink-0"
    >
      <div
        class="flex flex-col w-64"
      >
        <div
          class="flex flex-col h-0 flex-1"
        >
          <div
            class="flex items-center h-16 flex-shrink-0 px-4"
          >
            <div
              class="cursor-pointer"
            >
              <svg
                fill="none"
                height="48px"
                viewBox="0 0 48 48"
                width="48px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="white"
                  fill-opacity="0.01"
                  height="48"
                  width="48"
                />
                <circle
                  cx="24"
                  cy="22"
                  fill="#2F88FF"
                  r="4"
                  stroke="black"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="4"
                />
                <path
                  d="M21.5238 33.0539C22.7619 31.6487 25.2381 31.6486 26.4762 33.0539C27.7143 34.4592 26.4763 43.3608 25.6508 44.2975C24.8254 45.2342 23.1746 45.2342 22.3492 44.2975C21.5238 43.3608 20.2857 34.4591 21.5238 33.0539Z"
                  fill="#2F88FF"
                  stroke="black"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="4"
                />
                <path
                  d="M30.9668 30.513C33.4289 28.4957 35 25.4313 35 22C35 15.9249 30.0751 11 24 11C17.9249 11 13 15.9249 13 22C13 25.4313 14.5711 28.4957 17.0332 30.513"
                  stroke="black"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="4"
                />
                <path
                  d="M31.9258 38.1656C37.8928 35.2345 42 29.0969 42 22C42 12.0589 33.9411 4 24 4C14.0589 4 6 12.0589 6 22C6 29.0969 10.1072 35.2345 16.0742 38.1656"
                  stroke="black"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="4"
                />
              </svg>
            </div>
          </div>
          <div
            class="flex-1 flex flex-col overflow-y-auto"
          >
            <nav
              class="flex-1 px-2 py-4 space-y-1"
            >
              <div
                class="text-gray-700 hover:bg-gray-300 hover:text-gray-900 cursor-pointer group flex items-center px-2 py-2 text-base font-medium rounded-md"
              >
                <div
                  class="w-4 h-4"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-circle-play "
                    data-icon="circle-play"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M512 256c0 141.4-114.6 256-256 256S0 397.4 0 256S114.6 0 256 0S512 114.6 512 256zM188.3 147.1c-7.6 4.2-12.3 12.3-12.3 20.9V344c0 8.7 4.7 16.7 12.3 20.9s16.8 4.1 24.3-.5l144-88c7.1-4.4 11.5-12.1 11.5-20.5s-4.4-16.1-11.5-20.5l-144-88c-7.4-4.5-16.7-4.7-24.3-.5z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="px-2"
                >
                  Listen Now
                </div>
              </div>
              <div
                class="text-gray-700 hover:bg-gray-300 hover:text-gray-900 cursor-pointer group flex items-center px-2 py-2 text-base font-medium rounded-md"
              >
                <div
                  class="w-4 h-4"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-border-all "
                    data-icon="border-all"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M384 96V224H256V96H384zm0 192V416H256V288H384zM192 224H64V96H192V224zM64 288H192V416H64V288zM64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H384c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="px-2"
                >
                  Browse
                </div>
              </div>
            </nav>
            <section
              class="mb-10"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
